ledger/participant-integration-api/src/main/scala/platform/InMemoryState.scala

```diff
@@ -22,13 +22,14 @@ import scala.util.chaining._
 private[platform] class InMemoryState(
     val ledgerEndCache: MutableLedgerEndCache,
     val contractStateCaches: ContractStateCaches,
-    val transactionsBuffer: InMemoryFanoutBuffer,
+    val inMemoryFanoutBuffer: InMemoryFanoutBuffer,
     val stringInterningView: StringInterningView,
     val dispatcherState: DispatcherState,
 )(implicit executionContext: ExecutionContext) {
   private val logger = ContextualizedLogger.get(getClass)
+  @volatile private var _initialized = false
```

Comment:
why cant we just check whether dispatcher is running? Then we would not need an additional state here

Reply:
I would prefer to keep these two concepts separate:
1. The dispatcher can actually be not running (imagine an initialized indexer that instantly fails and re-triggers initialization).
2. It is harder to tie the two concepts for less informed readers as initialization is actually tied to what is happening before this `initializedTo` and has nothing to do with the Dispatcher. It is a bit confusing as it is and tie-ing it to the Dispatcher's lifecycle complicates things further.

Reply:
I understand, but still not sure it is worth to introduce an awkward state like this. This awfully seems to me like an optional mutable reference to a holder of mutable vals. Why do we need to make the holder of mutable val communicate when it is initialized? :thinking:

Reply:
I don't fully agree, but following your suggestion since it's a aesthetics decision.